### PR TITLE
fix: reduce per-system k_cutoff to shared max for batch Ewald Miller …

### PR DESCRIPTION
**Pull request by zubatyuk**
## Description

Fix incorrect broadcast of per-system `k_cutoff` tensor in batched Ewald summation. When `k_cutoff` is a 1-D tensor of shape `(B,)` and the batch size `B == 3`, element-wise multiplication against the 3-element reciprocal cell lengths silently produced per-axis cutoffs instead of a single shared scalar bound. This adds `_prepare_k_cutoff` to both Torch and JAX k-vector modules to reduce a per-system cutoff tensor to its maximum before computing Miller indices.

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Performance improvement
- [x] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] CI/CD or infrastructure change

## Changes Made

- Add `_prepare_k_cutoff` helper in `nvalchemiops/torch/interactions/electrostatics/k_vectors.py` that reduces a per-system `k_cutoff` tensor to its maximum scalar value before computing Miller bounds.
- Add matching `_prepare_k_cutoff` helper in `nvalchemiops/jax/interactions/electrostatics/k_vectors.py` with the same reduction logic.
- Update docstrings in `generate_miller_indices` / `_generate_miller_indices` and `generate_k_vectors_ewald_summation` to document the shared-max-cutoff behavior for batched inputs.
- Add tests for both Torch and JAX verifying that a per-system `k_cutoff` tensor produces the same k-vectors as passing the scalar maximum directly, including a `B=3` case that specifically catches the per-axis broadcast bug.
- Add batch auto-parameter Ewald tests for both bindings to verify end-to-end batched `alpha=None, k_cutoff=None` path.
- Add a note to the electrostatics user guide explaining how batch mode handles per-system `k_cutoff`.

## Testing

- [x] Unit tests pass locally (`make pytest`)
- [x] Linting passes (`make lint`)
- [x] New tests added for new functionality meets coverage expectations?

## Checklist

- [x] I have read and understand the [Contributing Guidelines](../CONTRIBUTING.md)
- [ ] I have updated the [CHANGELOG.md](../CHANGELOG.md)
- [x] I have performed a self-review of my code
- [x] I have added docstrings to new functions/classes
- [x] I have updated the documentation (if applicable)

## Additional Notes

The root cause is the element-wise multiply `k_cutoff * cell_lengths` in `generate_miller_indices` / `_generate_miller_indices`. When `k_cutoff` has shape `(B,)` and `cell_lengths` has shape `(3,)`, the result is `(B, 3)` via broadcasting, which silently gives per-axis bounds instead of per-system ones. The bug is latent for `B != 3` (shape mismatch triggers a broadcast error) but goes undetected when `B == 3`.

> [!TIP]
> This repository uses Greptile, an AI code review service, to help conduct
> pull request reviews. We encourage contributors to read and consider suggestions
> made by Greptile, but note that human maintainers will provide the necessary
> reviews for merging: Greptile's comments are **not** a qualitative judgement
> of your code, nor is it an indication that the PR will be accepted/rejected.
> We encourage the use of emoji reactions to Greptile comments, depending on
> their usefulness and accuracy.

### CODE REVIEW

File: test/interactions/electrostatics/bindings/torch/test_ewald.py [hunk 1]
```diff
@@ -4488,6 +4488,51 @@ def test_auto_estimate_alpha_and_k_cutoff(self, device):
...
+            .expand(2, -1, -1)
+            .contiguous()
+            * 10.0
+        )
+        batch_idx = torch.tensor([0, 0, 1, 1], dtype=torch.int32, device=device)
+        neighbor_list = torch.tensor(
+            [[0, 1, 2, 3], [1, 0, 3, 2]], dtype=torch.int32, device=device
+        )
+        neighbor_ptr = torch.tensor([0, 1, 2, 3, 4], dtype=torch.int32, device=device)
+        neighbor_shifts = torch.zeros((2, 3), dtype=torch.int32, device=device)
```
[the remark this answers is not in the document]
Comment (zubatyuk) [reply test/interactions/electrostatics/bindings/torch/test_ewald.py]: fixed

**Review by laserkelvin — APPROVED** [review]
Will approve - just have a minor conflict (just need to reinsert the doc text from #39) to resolve.

I'll merge after CI passes
